### PR TITLE
Tracing: Add new [tracing.opentelemetry] custom_attributes config setting

### LINKED ISSUE #54037: Unable to set custom attributes with new OTel tracing configuration
**What happened**:

In current versions of Grafana, using the [`tracing.jaeger`](https://grafana.com/docs/grafana/v9.0/setup-grafana/configure-grafana/#tracingjaeger) config section will cause Grafana to emit a warning:

```
logger=settings level=warn msg="[Deprecated] the configuration setting 'tracing.jaeger' is deprecated, please use 'tracing.opentelemetry.jaeger' instead"
```

However, the `tracing.opentelemetry.*` sections have no way to configure custom attributes (like the `always_included_tag` setting in `tracing.jaeger`).

Now, it is possible to define them with an environment variable ([`OTEL_RESOURCE_ATTRIBUTES`](https://opentelemetry.io/docs/reference/specification/sdk-environment-variables/#general-sdk-configuration)), but I'd rather use the config as we run Grafana with a clean environment.

**What you expected to happen**:

n/a

**How to reproduce it (as minimally and precisely as possible)**:

n/a

**Anything else we need to know?**:

I've already started working on a PR locally to add this - my instinct is that it might make sense to have a higher-level `tracing.opentelemtry` config section so that both the `jaeger` and `otlp` providers can use the same custom attributes configuration. Any strong feelings on this?

**Environment**:
- Grafana version: 9.0.x
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by hairyhenderson**
**What this PR does / why we need it**:

Adds a new config section `tracing.opentelemetry` with item `custom_attributes` to allow setting custom key:value pairs for OTel traces.

This is the equivalent of the `always_included_tag` setting in `tracing.jaeger`, but I named it `custom_attributes` since I thought that name describes a bit better what it does.

**Which issue(s) this PR fixes**:

Fixes #54037

**Special notes for your reviewer**:

_If possible_ I'd like to backport this to 9.1.x, but I'm not sure if adding new config items is allowed in a patch release...

Signed-off-by: Dave Henderson <dave.henderson@grafana.com>

### CODE REVIEW

**Comment by hairyhenderson** [comment]
I'm not sure why the Drone build is failing... 🤷‍♂️

**Comment by hairyhenderson** [comment]
🤔 The Drone build failures seem unrelated to this change... I think this is ready for review! 🙏

File: conf/defaults.ini
Comment (hairyhenderson) [reply conf/defaults.ini:1033]: > consistency and clarity

not sure what you mean by this? can you give me an example?

File: pkg/infra/tracing/opentelemetry_tracing.go
Comment (hairyhenderson) [reply pkg/infra/tracing/opentelemetry_tracing.go:99]: ah, this is a good idea... this is effectively a copy/paste from `splitTagSettings` in `tracing.go`, but IMO it'd be a worthy safety improvement. 👍

File: conf/defaults.ini
Comment (hairyhenderson) [reply conf/defaults.ini:1033]: OH, I see now, your comment was meant for the line under this 😅 - I typoed and had `custom_attribs` there instead of `custom_attributes`.

Comment (marefr) [reply conf/defaults.ini:1033]: Yes, sorry :+1:

File: pkg/infra/tracing/opentelemetry_tracing.go
Comment (hairyhenderson) [reply pkg/infra/tracing/opentelemetry_tracing.go]: Fixed

File: conf/defaults.ini
Comment (hairyhenderson) [reply conf/defaults.ini:1033]: Fixed! Also found the same error in `sample.ini` 😅

**Comment by hairyhenderson** [comment]
I've rebased, and addressed the review comments so far - PTAL 🙏

**Review by mildwonkey — APPROVED** [review]
I'm not sure how to confirm that the attributes are doing what you expect them to, but the code and parsing all LGTM. I'll let you decide if you need another approval or not. 

I like the error message on startup when the values are invalid, thanks for adding that! Not for this pr, but in the future we could add the top level section name (in this case `[tracing.opentelemetry]`) and the name of the file with the error.

File: pkg/infra/tracing/opentelemetry_tracing.go
Comment (bergquist) [inline pkg/infra/tracing/opentelemetry_tracing.go:128]: we are losing this one. is that on purpose?

Comment (hairyhenderson) [reply pkg/infra/tracing/opentelemetry_tracing.go:128]: this isn't being lost - it's refactored (see ln168)

File: pkg/infra/tracing/opentelemetry_tracing.go
Comment (idafurjes) [inline pkg/infra/tracing/opentelemetry_tracing.go:159]: I agree with the comment, tracing needs some more attention, I was not able to prioritize it so far

**Review by idafurjes — APPROVED** [review]
LGTM

**Pull request merged**
